### ISSUE TITLE
Validate model/adapter is available on the model server before sending requests to a model server

### ISSUE THREAD

**Issue by liu-cong** [kubernetes-sigs/gateway-api-inference-extension#49]
According to [Model Server Protocol](https://docs.google.com/document/d/18VRJ2ufZmAwBZ2jArfvGjQGaWtsQtAP6_yF2Xn6zcms/edit?tab=t.0#heading=h.sw2xdf66jh6), supported model servers should expose `AvailableModels`. In vLLM, the `/v1/models` endpoint exposes such info.

The ext proc should get the `AvailableModels` similar as scraping metrics, and add a filter to only pick model servers that have the targetModel in the request as available.

**Comment by hzxuzhonghu** [comment]
Is this already done

**Comment by ahg-g** [comment]
I don't think this is generally feasible on the long term, the server may have 100s of adapters.

**Comment by danehans** [comment]
Closing in favor of https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/1414
